backend/canisters/community/impl/src/updates/register_proposal_vote.rs

```diff
@@ -0,0 +1,133 @@
+use crate::activity_notifications::handle_activity_notification;
+use crate::{mutate_state, read_state, run_regular_jobs, RuntimeState};
+use canister_tracing_macros::trace;
+use chat_events::{MessageContentInternal, Reader, RecordProposalVoteResult};
+use community_canister::register_proposal_vote::{Response::*, *};
+use ic_cdk_macros::update;
+use types::{CanisterId, ChannelId, ProposalId, UserId};
+
+#[update]
+#[trace]
+async fn register_proposal_vote(args: Args) -> Response {
+    run_regular_jobs();
+
+    let PrepareResult {
+        user_id,
+        is_nns,
+        governance_canister_id,
+        proposal_id,
+    } = match read_state(|state| prepare(&args, state)) {
+        Ok(ok) => ok,
+        Err(response) => return response,
+    };
+
+    let c2c_args = user_canister::c2c_vote_on_proposal::Args {
+        is_nns,
+        governance_canister_id,
+        proposal_id,
+        adopt: args.adopt,
+    };
+    match user_canister_c2c_client::c2c_vote_on_proposal(user_id.into(), &c2c_args).await {
```

Comment:
The user canisters will need to be updated to allow votes from known communities

Reply:
Next PR